runbot_buildout/models/runbot_build.py

```diff
@@ -0,0 +1,434 @@
+                    version, ', '.join(buildout_branch.mapped('branch_name')),
+                )
+            )
+            self.write({
+                'state': 'done',
+                'result': 'ko',
+            })
+            self._github_status()
+        return buildout_build
+
+    @api.multi
+    def _init_from_buildout(self, buildout_build):
+        self.ensure_one()
+        if os.path.isdir(self._path()):
+            shutil.rmtree(self._path(), ignore_errors=True)
+        if not os.path.isdir(buildout_build._path()):
+            self._log(
+                'buildout',
+                'Buildout directory %s does not exist' % buildout_build._path()
+            )
+            return -2
+        self._log(
+            'buildout', 'Copying buildout from %s' % buildout_build._path()
+        )
+
+        def do_copy(buildout_path, own_path):
+            shutil.copytree(buildout_path, own_path, symlinks=True)
+            # we don't care about the logs created during the build of the
+            # buildout
+            shutil.rmtree(os.path.join(own_path, 'logs'), ignore_errors=True)
+            os.makedirs(os.path.join(own_path, 'logs'))
+
+        process = Process(
+            target=do_copy, args=(buildout_build._path(), self._path()),
+        )
+        process.run()
+        return process.pid
+
+    @api.multi
+    def _get_buildout_source_item(self):
+        self.ensure_one()
+        # the recipe switches working directory
+        cwd = os.getcwd()
+        buildout = Buildout(self._path('buildout.cfg'), [])
+        section = buildout[self.buildout_section]
+        recipe = BaseRecipe(buildout, 'buildout', section)
+        recipe.parse_addons(section)
+        os.chdir(cwd)
+
+        for parts_dir, (loc_type, loc_spec, options) in recipe.sources.items():
+            # we only support git urls
+            if loc_type != 'git':
+                continue
+            if self.repo_id.name == loc_spec[0]:
+                return parts_dir, (loc_type, loc_spec, options)
+
+    @api.multi
+    def _get_buildout_parts_path(self):
+        item = self._get_buildout_source_item()
+        if item:
+            return item[0]
+
+    @api.multi
+    def _adapt_buildout(self):
+        self.ensure_one()
+
+        path, dummy = self._get_buildout_source_item()
+
+        if not path:
+            self._log(
+                'buildout', 'No addons line found for %s' % self.repo_id.name,
+            )
+            self.write({
+                'state': 'done',
+                'result': 'ko',
+            })
+            self._github_status()
+            return False
+
+        with tempfile.NamedTemporaryFile(
+                prefix='buildout', suffix='.cfg', dir=self._path(),
+                delete=False,
+        ) as buildout_file:
+            buildout_file.write(bytes(
+                '[buildout]\n'
+                'extensions = buildout.wheel\n'
```

Comment:
Probably we then also have to change this?

Reply:
yes, and I'm going to ruhtlessly steal your snippet to use newer setuptools